### PR TITLE
Feature/separate explore deep content

### PR DESCRIPTION
**Pull request by AdityaKhatri**
## Changes

* Separate explore deep content

Depends on: https://github.com/the-deep/deeper/pull/981
Depends on: https://github.com/the-deep/server/pull/1272

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] codegen errors
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

## This PR contains valid:

- [x] permission checks
- [x] translations

### CODE REVIEW

File: app/views/ExploreDeepContent/BrushLineChart.tsx [hunk 1]
```diff
@@ -41,6 +46,8 @@ function BrushLineChart(props: Props) {
         [startDate, endDate, maxCount],
     );
 
+    console.warn(selectedDomain);
+
```
Comment (tnagorra) [inline app/views/ExploreDeepContent/BrushLineChart.tsx]: Let's remove console :D

File: app/Base/configs/env.ts [hunk 2]
```diff
@@ -10,6 +10,7 @@ export const isDebugMode = process.env.REACT_APP_DEBUG_MODE?.toLowerCase() === '
 export const apiHttps = process.env.REACT_APP_API_HTTPS || 'http';
 export const apiEndpoint = process.env.REACT_APP_API_END as string; // localhost:8000
 export const graphqlEndpoint = process.env.REACT_APP_GRAPHQL_ENDPOINT as string; // http://localhost:8000/graphql
+export const staticEndpoint = process.env.REACT_APP_STATIC_ENDPOINT as string; // http://localhost:8000/graphql
```
Comment (tnagorra) [inline app/Base/configs/env.ts]:
```suggestion
export const staticEndpoint = process.env.REACT_APP_STATIC_ENDPOINT as string;
```

File: app/views/PublicExploreDeep/index.tsx [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+        setStartDate,
+    ] = useState<number | undefined>(
+        () => lastYearDateTime,
+    );
+    const [
+        endDate = todaysDateTime,
+        setEndDate,
+    ] = useState<number | undefined>(
+        () => todaysDateTime,
+    );
```
Comment (tnagorra) [inline app/views/PublicExploreDeep/index.tsx]:
```suggestion
    const [
        startDate = deepStartDateTime,
        setStartDate,
    ] = useState<number | undefined>(
        lastYearDateTime,
    );
    const [
        endDate = todaysDateTime,
        setEndDate,
    ] = useState<number | undefined>(
        todaysDateTime,
    );
```

File: app/views/PublicExploreDeep/index.tsx [hunk 4]
```diff
@@ -0,0 +1,368 @@
...
+                    (foo, bar) => compareNumber(foo.year, bar.year),
+                );
+                if (latestYear) {
+                    setSelectedYear(latestYear.id);
+                    const yearStartDate = new Date(latestYear.year, 0, 1).getTime();
+                    const yearEndDate = new Date(latestYear.year, 11, 31).getTime();
+                    setStartDate(yearStartDate);
+                    if (yearEndDate < endDate) {
+                        setEndDate(yearEndDate);
+                    }
```
Comment (tnagorra) [inline app/views/PublicExploreDeep/index.tsx]: Isn't this supposed to be setEndDate(Math.min(yearEndDate, endDate))

File: app/views/PublicExploreDeep/index.tsx [hunk 5]
```diff
@@ -0,0 +1,368 @@
...
+    }, [
+        csvDownloadLink,
+        csvDownloadName,
+    ]);
+
+    const snapshotOptions = useMemo(() => (
+        data?.publicDeepExploreYearlySnapshots?.filter((snapshot) => snapshot.year !== 0)
+    ), [data?.publicDeepExploreYearlySnapshots]);
+
+    const onSnapshotChange = useCallback((newSnapshotId: string) => {
```
Comment (tnagorra) [inline app/views/PublicExploreDeep/index.tsx]:
```suggestion
    const handleSnapshotChange = useCallback((newSnapshotId: string) => {
```

File: app/views/ExploreDeepContent/index.tsx [hunk 6]
```diff
@@ -0,0 +1,489 @@
...
+                                projectsByRegion={projectsByRegion ?? undefined}
+                                readOnly={printPreviewMode || isPublic}
+                                endDate={endDate}
+                                startDate={startDate}
+                                onEndDateChange={!isPublic ? onEndDateChange : undefined}
+                                onStartDateChange={!isPublic ? onStartDateChange : undefined}
+                                projectFilters={filters}
+                                isPublic={isPublic}
+                                completeTimeseries={projectCompleteTimeseries ?? undefined}
+                            />
```
Comment (tnagorra) [inline app/views/ExploreDeepContent/index.tsx]: We can re-use these

File: app/views/ExploreDeepContent/index.tsx [hunk 7]
```diff
@@ -0,0 +1,489 @@
...
+                                endDate={endDate}
+                                startDate={startDate}
+                                readOnly={printPreviewMode || isPublic}
+                                onEndDateChange={!isPublic ? onEndDateChange : undefined}
+                                onStartDateChange={!isPublic ? onStartDateChange : undefined}
+                                completeTimeseries={entriesCompleteTimeseries ?? undefined}
+                                leadsCountByDay={leadsCountByDay ?? undefined}
+                                entriesCountByRegion={entriesCountByRegion ?? undefined}
+                                loading={loadingMapAndLeadData}
+                            />
```
Comment (tnagorra) [inline app/views/ExploreDeepContent/index.tsx]: We can re-use these

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: app/views/ExploreDeepContent/BrushLineChart.tsx (hunk 1 above)
Comment (AdityaKhatri) [reply app/views/ExploreDeepContent/BrushLineChart.tsx]: @tnagorra , lets remove console :D
image

Comment (tnagorra) [reply app/views/ExploreDeepContent/BrushLineChart.tsx]: Yes.

File: app/views/ExploreDeepContent/BrushLineChart.tsx [hunk 8]
```diff
@@ -68,20 +73,25 @@ function BrushLineChart(props: Props) {
...
             </svg>
-            <VictoryGroup
+            <VictoryChart
                 // Need to dismount chart when width changes
                 // https://github.com/FormidableLabs/victory/issues/1173
                 key={width}
                 width={width}
                 height={height}
                 scale={{ x: 'time', y: 'linear' }}
+                // FIXME: this should be unset on public dashboard
```
Comment (tnagorra) [inline app/views/ExploreDeepContent/BrushLineChart.tsx]: Please check for maxDomain.
For public dashboard (for a year), the minDomain and maxDomain should be the first second of the year and the last second of the year respectively.

File: app/views/PublicExploreDeep/index.tsx [hunk 9]
```diff
@@ -0,0 +1,393 @@
...
+                const latestYear = getMaximum(
+                    response.publicDeepExploreYearlySnapshots,
+                    (foo, bar) => compareNumber(foo.year, bar.year),
+                );
+                if (latestYear) {
+                    setSelectedYear(latestYear.id);
+                    const yearStartDate = new Date(latestYear.startDate).getTime();
+                    const yearEndDate = new Date(latestYear.endDate).getTime();
+                    setStartDate(yearStartDate);
+                    setEndDate(Math.min(yearEndDate, endDate));
```
Comment (tnagorra) [inline app/views/PublicExploreDeep/index.tsx]: If we are reading startDate and endDate from `                    response.publicDeepExploreYearlySnapshots`, shouldn't we also get maximum comparing those values

File: app/views/ExploreDeepContent/index.tsx [hunk 10]
```diff
@@ -0,0 +1,503 @@
...
+        exportCreatePending,
+        filters,
+        loading,
+    } = props;
+
+    const startDateString = formatDateToString(new Date(startDate), 'yyyy-MM-dd');
+    const endDateString = formatDateToString(new Date(endDate), 'yyyy-MM-dd');
+
+    const [representationType, setRepresentationType] = useState<Option['key']>('table');
+    //
```
Comment (tnagorra) [inline app/views/ExploreDeepContent/index.tsx]:
```suggestion

```

File: app/views/ExploreDeepContent/index.tsx [hunk 11]
```diff
@@ -0,0 +1,503 @@
...
+        endDate,
+        startDate,
+        printPreviewMode,
+        isPublic,
+        onEndDateChange,
+        onStartDateChange,
+        entriesCompleteTimeseries,
+        leadsCountByDay,
+        entriesCountByRegion,
+    ]);
```
Comment (tnagorra) [inline app/views/ExploreDeepContent/index.tsx]: Why separate out the props? Why not directly define variables for ProjectContent and PEntriesContent component

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: app/views/ExploreDeepContent/BrushLineChart.tsx (hunk 8 above)
Comment (AdityaKhatri) [reply app/views/ExploreDeepContent/BrushLineChart.tsx]: I think this is fine.

**Review by tnagorra — APPROVED** [review]

**Pull request merged**